### ISSUE TITLE
Support for X-52 Stick and Throttle

### ISSUE THREAD

**Issue by mdjarv** [mdjarv/JoystickVisualizer#16]
X-52 stick and throttle are connected as a single USB device

These are the axis:

> Y for Throttle
> X for Roll
> Y for Pitch
> Z for Yaw

See #14 for more details

**Comment by mdjarv** [comment]
Since the X-52 HOTAS show up as a single device we need to figure out how to show only the device(s) being used.

Check if we can enable the throttle only if axis Z is triggered, and the stick only if axis X/Y is triggered, that way these devices can share USB ID but work independently

**Comment by brittz** [comment]
> Check if we can enable the throttle only if axis Z is triggered, and the stick only if axis X/Y is triggered, that way these devices can share USB ID but work independently

I think it should be work on that way.

In the Proxy I receive all inputs, but some diferent as I described above (X-Planet setup). See:
**Throttle**: 06a3:075c,Saitek X52 Throttle,Z=65535
**Stick up and down**: 06a3:075c,Saitek X52 Throttle,X=30655
**Stick left and right**: 06a3:075c,Saitek X52 Throttle,Y=31999
**Stick Yaw**: 06a3:075c,Saitek X52 Throttle,RotationZ=63158

In the Visualizer, it remaing waiting for device until I move the Stick or Throttle, after that, the 3D Throttle model shows up and start working.

**Comment by mdjarv** [comment]
Could you test release 0.9.8 and see if both the throttle and joystick works

https://github.com/mdjarv/JoystickVisualizer/releases/tag/v0.9.8

**Comment by brittz** [comment]
Tested and worked 100%! Thank you!!!

**Comment by SilverlightPony** [comment]
Not working for me.  Maybe a difference between basic X52 and X52 Pro?

X52 Pro USB hardware IDs:  https://i.imgur.com/3nzMwQn.png

Axes:

- X and Y - stick left/right and forward/back (i.e., usually roll and pitch)  respectively
- Z - throttle
- Z-rotation - stick twist (usually yaw)
- X-rotation - lower throttle rotary knob
- Y-rotation - upper throttle rotary knob
- slider - throttle thumb slider

**Comment by mdjarv** [comment]
@brittz 
Nice! One of these days I might even be able to make a model for it with enough practice.

@SilverlightPony
Yeah, the Pro had a different ID so thanks for getting it for me. I'll add it and make a new release in a few minutes

**Comment by mdjarv** [comment]
@SilverlightPony 
How about this?
https://github.com/mdjarv/JoystickVisualizer/releases/tag/v0.9.9

**Comment by SilverlightPony** [comment]
Reading throttle position.  Stick is not shown, but I get the feeling that's intentional (or at least a known limitation at present).  Now if only you had the correct 3D model.  :P

Lemme know if you need any more info for improvements.

**Comment by mdjarv** [comment]
Hmm, stick should show if you move it, just like the throttle should show when you move it... just to be clear, you are using both X52 Pro throttle and stick?

**Comment by SilverlightPony** [comment]
Ah, yup.  It's not animating stick-twist tho.

**Comment by mdjarv** [comment]
Yeah, unfortunately adding proper twist movement is quite difficult as it requires a mix of local and global rotations to get right (I've done it once and it took a few more hours than I'd like to admit).

Maybe if we get a proper 3D model it would be worth the effort but the Warthog model is a bit too cumbersome to work with in this regard.

**Comment by Blu3wolf** [comment]
Just wanted to point out there are X-52 Pros which have the PID 0255 - I believe anyone with one of those sticks would not be supported currently?
